cmake/everest-generate.cmake

```diff
@@ -97,7 +95,14 @@ function(ev_add_project)
     if (EXISTS "${MODULES_DIR}/CMakeLists.txt")
         # FIXME (aw): default handling of building all modules?
         if (EVC_MAIN_PROJECT OR NOT EVEREST_DONT_BUILD_ALL_MODULES)
+            # We need a list of all to-be-built rust modules to create our
+            # workspace Cargo.toml, but we cannot easily pass it through the
+            # ev_add_module calls without polluting all CMakeLists.txt files
+            # with it. We bypass this using a global.
+            set_property(GLOBAL PROPERTY RUST_MODULE_LIST "")
```

Comment:
Not sure about this one, but doesn't that reset the list all the time?

Reply:
Furthermore, unfortunately there is this global variable, but a bit cleaner might be to create a custom target with `add_custom_target` called something like `everest_core_build_rust_modules`, then, you can set a property on this target with `set_property( target ...)` (so we have at least some 'encapsulation')
furthermore, when running the `add_rust_module ...`, this function can add additional targets/dependencies to the `everest_core_build_rust_modules` if necessary, so we have proper dependency tracking

Reply:
I defer here, not sure how to implement that.